common/src/main/java/com/scottlogic/deg/common/date/ChronoUnitWorkingDayWrapper.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.common.date;
+
+import java.time.temporal.ChronoUnit;
+
+public class ChronoUnitWorkingDayWrapper {
```

Comment:
I agree mostly with this. I'm a bit adverse to entirely removing the wrapper and passing around the adjuster because this would necessitate all of the logic being held in the constraint reader map, which I believe is too early in the process to do such processing.

Reply:
if still want to wrap it so that  have a class that deals with creating the adjuster (rather than putting it in the EqualToFieldReader class)  then maybe have a small hierarchy where one subclass implements the working day version and one subclass implements the chrono unit version of the interface.  I think that's better than mixing both in same wrapper class.  would need factory with 2 create methods like i suggested above for temporalAdjuster.  Also rather than call the interface ChronoUnitWorkingDayWrapper it would now be called something like DateTimeAdjustmentUnit or TemporalAdjusterFactory.

I would then construct and call adjuster() on this instrance within EqualTofieldReader code.   I'd still put TemporalAdjuster in the Constraint rather than a class than has an adjuster() method to create a TemporalAdjuster.

Reply:
@afroggattsl Let me know if you're happy with this current implementation, or if you want me to split it out into two classes.